src/vsr/replica.zig

```diff
@@ -2327,9 +2333,17 @@ pub fn ReplicaType(
             }
 
             if (self.syncing == .updating_superblock) {
+                // State sync can "truncate" the first batch of committed ops!
                 maybe(self.commit_min >
```

Comment:
TIL that commit_min can actually regress during state sync! It somehow didn't register with me before, but looks correct: you might get stuck compacting the last batch of checkpoint's prepares

Reply:
:laughing: I was surprised by the same thing a while back -- state_checker has a workaround for it!: https://github.com/tigerbeetle/tigerbeetle/blob/aea9cbf1391768c78f35fe31988187119383903e/src/testing/cluster/state_checker.zig#L101-L107